### PR TITLE
octave: switch to qt6, fix cross-build

### LINKED ISSUE #51602: Could not plot with GNU/Octave on aarch64 machine
### Is this a new report?

Yes

### System Info

Void 6.6.42_1 aarch64

### Package(s) Affected

octave-9.2.0_1

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

_No response_

### Expected behaviour

For example, when I plot a single point by `plot(0,0)` in octave-cli (on raspberry pi 5), it should pop-out figure window and returns nothing.

### Actual behaviour

For example, when I plot a single point by `plot(0,0)` in octave-cli (on raspberry pi 5), it returns 

"The FLTK graphics toolkit is not actively maintained and has a number of limitations that are unlikely to be fixed. The qt toolkit is recommended instead.", "Insufficient GL support", and "error: unable to plot due to insufficient OpenGL support". 

However, The plotting of octave is working correctly in raspberry pi offical OS (octave use qt toolkit). Both vedio driver are "vc4-kms-v3d", so I don't think that hardware is not support. What's wrong with that?

I also found that octave-gui could not launch, the error message is 

"octave: GUI features missing or disabled in this build"

I do not know whether plotting problem is related to this (I found "qt" is not exist in octave command `available_graphic_toolkits()` might related to built without Qt support).

### Steps to reproduce

1. Open octave CLI 2. plot with `plot(0,0)`

### PR DESCRIPTION
**Pull request by tornaria**
#### Testing the changes
- I tested the changes in this PR: **NO**

I cross built the package for aarch64 before and after and the log shows clearly that qt5 was broken before and claims to be working now. But I don't have an aarch64 box to actually test if this fixes #51602.

Closes: #51602 (hopefully)

### CODE REVIEW

File: srcpkgs/octave/template [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'octave'
 pkgname=octave
 version=9.2.0
-revision=1
+revision=2
 build_style=gnu-configure
 configure_args="--disable-docs"
 hostmakedepends="perl gcc-fortran pkg-config"
```
Comment (classabbyamp) [inline srcpkgs/octave/template:7]:
```suggestion
 hostmakedepends="perl gcc-fortran pkg-config  $(vopt_if gui qt5-host-tools)"
```

Comment (tornaria) [reply srcpkgs/octave/template:7]: Sure, it's the same. I'm just following the style of the existing template (which I did not originally write).

Comment (classabbyamp) [reply srcpkgs/octave/template]: would rather follow what most templates do than what one person did like 10 years ago

Comment (tornaria) [reply srcpkgs/octave/template]: It's done. Still waiting for someone with aarch64 to confirm this fixes #51602.

**Comment by sgn** [comment]
I would move this to Qt6, too!

**Comment by tornaria** [comment]
This seems to be working fine now, with qt6. Can someone test the cross built pkg on aarch64?

@dkwo @a44881212


I will squash later. I'm trying to update hdf5 and see if it can be cross compiled.

**Comment by dkwo** [comment]
If the warnings on crosscompile are gone, I'd suggest to proceed: It will most likely be better than present (broken) situation.
I won't have access to aarch64 for a few more weeks, but I'll test it (and sage) later on.

File: srcpkgs/GraphicsMagick/template [hunk 2]
```diff
@@ -45,7 +45,7 @@ libgraphicsmagick-devel_package() {
 		vmove "usr/lib/*.so"
 		vmove "usr/lib/*.la"
 		vmove "usr/share/man/man1/*-config.1"
-		vcopy ${wrksrc}/PerlMagick usr/share/doc/GraphicsMagick*/
+		vcopy ${wrksrc}/PerlMagick usr/share/doc/GraphicsMagick/
```
Comment (classabbyamp) [inline srcpkgs/GraphicsMagick/template:48]:
```suggestion
		vcopy PerlMagick usr/share/doc/GraphicsMagick/
```
PWD should be $wrksrc already

Comment (tornaria) [reply srcpkgs/GraphicsMagick/template]: Good catch, done. Only change in the force-push.

**Comment by dkwo** [comment]
@tornaria The crosscompiled octave has a working gui, as expected. Thanks!

**Pull request merged**